com.zsmartsystems.zigbee/src/main/java/com/zsmartsystems/zigbee/zcl/ZclCluster.java

```diff
@@ -660,11 +665,32 @@ public void sendDefaultResponse(Integer transactionId, Integer commandIdentifier
      * @param transactionId the transaction ID to use in the response
      * @param commandIdentifier the command identifier to which this is a response
      * @param status the {@link ZclStatus} to send in the response
+     * @deprecated use sendDefaultResponse(ZclCommand command, ZclStatus status)
      */
+    @Deprecated
     public void sendDefaultResponse(Integer transactionId, Integer commandIdentifier, ZclStatus status) {
         sendDefaultResponse(transactionId, commandIdentifier, status, null);
     }
 
+    /**
+     * Sends a default response to the client
+     *
+     * @param command the {@link ZclCommand} to which we are responding
+     * @param status the {@link ZclStatus} to send in the response
+     */
+    public void sendDefaultResponse(ZclCommand command, ZclStatus status) {
+        ZclCommand response = createDefaultResponse(command, status);
+        if (response == null) {
+            return;
+        }
+        if (isManufacturerSpecific()) {
+            response.setManufacturerCode(getManufacturerCode());
+        }
```

Comment:
I the now deprecated method there is also an `else` branch. Are you setting the `manufacturerCode` value somewhere else now?

```
if (isManufacturerSpecific()) {
            defaultResponse.setManufacturerCode(getManufacturerCode());
        } else if (manufacturerCode != null) {
            defaultResponse.setManufacturerCode(manufacturerCode);
        }
```

Reply:
Yes - in the ```createDefaultResponse``` it handles the check if the command has the manufacturer specific flag set. This cluster specific check can't be done in that method as it's a static helper method that's also called from the ```ZigBeeNode``` and ```ZigBeeEndpoint```.